packages/data-connect/src/core/AppCheckTokenProvider.ts

```diff
@@ -29,10 +30,14 @@ import { Provider } from '@firebase/component';
  */
 export class AppCheckTokenProvider {
   private appCheck?: FirebaseAppCheckInternal;
+  private serverAppAppCheckToken?: string;
   constructor(
-    private appName_: string,
+    app: FirebaseApp,
     private appCheckProvider?: Provider<AppCheckInternalComponentName>
   ) {
+    if (_isFirebaseServerApp(app) && app.settings.appCheckToken) {
+      this.serverAppAppCheckToken = app.settings.appCheckToken;
+    }
     this.appCheck = appCheckProvider?.getImmediate({ optional: true });
```

Comment:
During initialization of the `AppCheckTokenProvider` that Data Connect implements, check to see if the provided instance of `FirebaseApp` is one of `FirebaseServerApp`, and if it contains an App Check token, then store it locally so that it can be returned by this provide when its `getToken` method is invoked (below).